File: GraffiTab-iOS/Classes/ViewControllers/Main/Main/Search/SearchUsersViewController.m
//
//  SearchUsersViewController.m
//  GraffiTab-iOS
//
//  Created by Georgi Christov on 14/04/2015.
//  Copyright (c) 2015 GraffiTab. All rights reserved.
//

#import "SearchUsersViewController.h"

@interface SearchUsersViewController ()

@end

@implementation SearchUsersViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)setSearchString:(NSString *)searchString {
    _searchString = searchString;
    
    [self refresh];
}

- (void)loadItems:(BOOL)isStart withOffset:(int)o successBlock:(void (^)(GTResponseObject *))successBlock cacheBlock:(void (^)(GTResponseObject *))cacheBlock failureBlock:(void (^)(GTResponseObject *))failureBlock {
    if (!self.searchString || self.searchString.length <= 0) {
        [GTUserManager getMostActiveUsersWithStart:o numberOfItems:MAX_ITEMS useCache:isStart successBlock:^(GTResponseObject *response) {
            successBlock(response);
        } cacheBlock:^(GTResponseObject *response) {
            cacheBlock(response);
        } failureBlock:^(GTResponseObject *response) {
            failureBlock(response);
        }];
    }
    else {
        // Perform search.
        [GTSearchManager searchUsersWithQuery:self.searchString offset:o numberOfItems:MAX_ITEMS successBlock:^(GTResponseObject *response) {
            successBlock(response);
        } failureBlock:^(GTResponseObject *response) {
            failureBlock(response);
        }];
    }
}

@end

File: GraffiTab/Classes/ViewControllers/Home/Search/SearchUsersViewController.swift
//
//  SearchUsersViewController.swift
//  GraffiTab
//
//  Created by Georgi Christov on 12/04/2016.
//  Copyright © 2016 GraffiTab. All rights reserved.
//

import UIKit
import GraffiTab_iOS_SDK

class SearchUsersViewController: ListUsersViewController {

    var searchQuery: String?
    
    override func viewWillAppear(animated: Bool) {
        super.viewWillAppear(animated)
        
        // Register analytics events.
        AnalyticsUtils.sendScreenEvent(self)
    }
    
    func search(query: String) {
        pullToRefresh.startRefreshing()
        searchQuery = query
        
        refresh()
    }
    
    // MARK: - ViewType-specific helpers
    
    override func getPadding(spacing: CGFloat) -> CGFloat {
        if DeviceType.IS_IPAD {
            return super.getPadding(spacing)
        }
        return 7
    }
    
    // MARK: - Loading
    
    override func loadItems(isStart: Bool, offset: Int, cacheBlock: (response: GTResponseObject) -> Void, successBlock: (response: GTResponseObject) -> Void, failureBlock: (response: GTResponseObject) -> Void) {
        if searchQuery == nil || searchQuery?.characters.count <= 0 {
            GTUserManager.getMostActive(offset, cacheResponse: isStart, cacheBlock: cacheBlock, successBlock: successBlock, failureBlock: failureBlock)
        }
        else {
            GTUserManager.search(searchQuery!, offset: offset, cacheResponse: isStart, cacheBlock: cacheBlock, successBlock: successBlock, failureBlock: failureBlock)
        }
    }
}
